compiler/src/main/scala/org/scalajs/nscplugin/GenJSCode.scala

```diff
@@ -743,7 +743,7 @@ abstract class GenJSCode[G <: Global with Singleton](val global: G)
     def genAnonJSClassNew(sym: Symbol, jsSuperClassValue: js.Tree,
         args: List[js.Tree])(
         implicit pos: Position): js.Tree = {
-      assert(sym.isAnonymousClass && !isJSFunctionDef(sym),
+      assert(isAnonymousJSClass(sym) && !isJSFunctionDef(sym),
```

Comment:
Isn't `&& !isJSFunctionDef(sym)` redundant, now?

If it isn't, why is there `|| isJSFunctionDef(sym)` above?